### ISSUE TITLE
SIM118 fixes a case in __contains__ which was using self, causing recusrion

### ISSUE THREAD

**Issue by HexDecimal** [astral-sh/ruff#6163]
An issue similar to #4481, Ruff incorrectly marks a case of [SIM118](https://beta.ruff.rs/docs/rules/in-dict-keys/) as an error and as fixable when neither is the case.

```py
from typing import KeysView


class Foo:
    def keys() -> KeysView[object]:
        ...

    def __contains__(self, key: object) -> bool:
        return key in self.keys()  # SIM118 replaces this with `return key in self`
```
The above is a likely pattern when writing a subclass of `collections.abc.Mapping`.

Command: `ruff example.py --select SIM --fix`
Version: `ruff 0.0.280`

**Comment by charliermarsh** [comment]
We can make the rule ignore self.keys which will almost always be wrong, but that’s probably the best we can do at present.

**Comment by charliermarsh** [comment]
Or ignoring this rule in __contains__ would also be reasonable.

**Comment by HexDecimal** [comment]
I'm not in a hurry. I'm already ignoring these manually and checked to see if there was a issue for this.

Ignoring `self.keys()` in `__contains__` might work.  Since this issue is very specific.
